opm/autodiff/WellInterface_impl.hpp

```diff
@@ -979,6 +979,42 @@ namespace Opm
 
 
 
+
+    template<typename TypeTag>
+    bool
+    WellInterface<TypeTag>::isVFPActive() const
+    {
+        // since the well_controls only handles the VFP number when THP constraint/target is there.
+        // we need to get the table number through the parser, in case THP constraint/target is not there.
+        // When THP control/limit is not active, if available VFP table is provided, we will still need to
+        // update THP value. However, it will only used for output purpose.
+
+        if (well_type_ == PRODUCER) { // producer
+            const int table_id = well_ecl_->getProductionProperties(current_step_).VFPTableNumber;
+            if (table_id <= 0) {
+                return false;
+            } else {
+                if (vfp_properties_->getProd()->getTable(table_id)) {
+                    return true;
+                }
+            }
+
+        } else { // injector
+            const int table_id = well_ecl_->getInjectionProperties(current_step_).VFPTableNumber;
+            if (table_id <= 0) {
+                return false;
+            } else {
+                if (vfp_properties_->getInj()->getTable(table_id)) {
```

Comment:
I think this `if` should also be removed, the code should just be:
```
return table_id > 0;
```
(also in production part).

Reply:
Then we will miss the `throw` .

Reply:
Or, we get a `throw` in the later calculation.  I do not mind either and I do not think the current form is bad.

Reply:
Indeed, it will throw from somewhere else. But calling a method to see if it throws is not a good approach. If we want to dig deeper here (I do not think we should) we should add a haveTable() method we could call, and make a nice message here, but for the current use, it would be overengineering.

Reply:
I think this is a nice point. Let me do that. It should be quick.